flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesHighAvailabilityTestBase.java

```diff
@@ -110,6 +117,11 @@ public void teardown() throws Exception {
 
         Context() {
             flinkKubeClient = createFlinkKubeClient();
+            configMapSharedWatcher =
+                    flinkKubeClient.createConfigMapSharedWatcher(
+                            KubernetesUtils.getConfigMapLabels(
+                                    CLUSTER_ID, LABEL_CONFIGMAP_TYPE_HIGH_AVAILABILITY));
+            watchExecutorService = Executors.newCachedThreadPool();
```

Comment:
Should we manage this service the same way as `org.apache.flink.kubernetes.highavailability.KubernetesHighAvailabilityTestBase#executorService`?